### ISSUE TITLE
UI/UX in Add Importer flow

### ISSUE THREAD

**Issue by baruchiro** [baruchiro/israeli-bank-scrapers-desktop#49]
When we adding an Importer (Scraper), we need to meet the following requirements:

- [ ] When the user clicks on `Add`, the Importer should appear on the screen. Currently, the Importer added under the "Add Scraper" sub-menu, so when the user clicks "Add", they see no effect (except that the fields are being empty).
- [ ] Consider whether to collapse the current "Add Scraper", or the whole sub-menu, after adding an Importer.
- [ ] Paging in the main table.

**Comment by ChenAlon** [comment]
I played with the flow/UI/UX.
The code is rough and as you can see does integrate with the actual data but I think it looks a bit nicer and friendly although it's very initial :)

I suggest the following:

## First-time user / Add another importer screen
* The user only sees cards for each importer  
  
![image](https://user-images.githubusercontent.com/4798088/73403505-395de980-42f8-11ea-8193-86b7a4d1cb80.png)
* User clicks on "Add Importer", a dialog window opens with the relevant inputs  
  
![image](https://user-images.githubusercontent.com/4798088/73403584-68745b00-42f8-11ea-9757-d48f3d2cc16c.png)

## Added importers screen
> The user must have at least one added Importer to be able to navigate to this screen
* List of added importers as a left sidebar
* The data table on the right
* Fixed export button on the right bottom with popover to select the export format
  
![image](https://user-images.githubusercontent.com/4798088/73405554-e0448480-42fc-11ea-87e0-99fe1df71d7e.png)

**Comment by baruchiro** [comment]
Hi, welcome to the project, and **thank you**!

Any issue is welcome, but especially a detailed comment that will take us forward, so thank you again for your work.

> The code is rough

For you, but for me, it is already a wonderful design, compared to what we have now.

-----

Now, to the practics:

1. I have to tell you that despite the great improvement you offer, we try to finish what is necessary for the [beta](https://github.com/baruchiro/israeli-bank-scrapers-desktop/issues?q=is%3Aopen+is%3Aissue+milestone%3ABeta) first. 
The intent is to focus on the minimal things that allow the **simple user** to use the app. I agree that **UI\UX is important**, so we will try to implement the **critical principles** in the beta, but the issue of replacing the (ugly) current design will have to wait.
2. But if you want to do it yourself today, of course you're welcome.
3. **I think we should to it in steps, that is why we may want to decide what the steps are before starting to implement.**
4. About the "Exporting" design: 
As I said, the [Minimal Valuable Flow](https://github.com/baruchiro/israeli-bank-scrapers-desktop#welcome-to-the-beta) is to **import** and **export** the data. 
There are many data analysis tools, of which there are many specifically designed for expense analysis, so I don't see the need for further analysis tools at this time. 
Therefore, the purpose of our tool right now is to **import** and **export** the expenses. Exporting is an important part of the software, and I would like to add more export methods as we think about them and as users request.
Therefore, I think the export area 
    - needs to be **larger**, 
    - and designed to contain **more than two exporters**, 
    - and to support exporters that require **additional settings** (as they already exist)

-----

And again,

## Thank you!

**Comment by ChenAlon** [comment]
Sure, I understand.
I can do it myself, most of the code is already there, but I think it's better to complete the beta issues first, as you said.

I'll see where I can help with the beta issues.

Great project, keep it up!

**Comment by baruchiro** [comment]
I added a `Beta` milestone to this issue.
We're getting closer to the Release, and I think we have some UI improvements, especially at the points mentioned above.

If anyone thinks there are improvements that need to be made before the Release, please add here.

**Comment by baruchiro** [comment]
About the **main table pagination**, I can't see how to [simply implement](https://github.com/ElemeFE/element/issues/6373) this with our [ElementUI table](https://element.eleme.io/#/en-US/component/table), and support sorting (and filtering in the future).

In contrast, I see [Vutify](https://vuetifyjs.com/en/components/data-tables) come with pagination out of the box. The reason I chose the ElementUI library was that it has the most stars in Github, or something. So if you think [Vuetify is better](https://github.com/ElemeFE/element/issues/6373#issuecomment-422905524), please let me know.

**Comment by baruchiro** [comment]
Hi @ChenAlon,
@Arielgordon123 and I decided to move from `element-ui` to `Vuetify`, mainly because their `DataTable` is more suited to our needs, but also because it is probably a better and more customizable library.

We know that some programmers claim that the library is heavy and that we can write the CSS ourselves, but we don't want to mess with CSS too much.

I am currently focusing on the porting of necessary components (`DataTable` and `Main.vue`, for example), but we don't want to maintain two libraries, so we will move to Vuetify over time.

We would love to hear your opinion, since you presented a very beautiful design.

**Comment by ChenAlon** [comment]
Hi guys,

Vuetify is awesome, great choice! 👌
I worked with it on a couple of projects, since version 0.7.x.  

I'm not sure about the "heaviness" and if it's heavier than others.
Eventually, even if it's heavier, it worth it as it gives you tons of components and flexibility that I don't think others give.

Let me know where you want me to contribute and I'll do my best to find time to work on it next week.

**Comment by baruchiro** [comment]
@ChenAlon I created a [new issue](https://github.com/baruchiro/israeli-bank-scrapers-desktop/issues/98) to synchronize the work between us, so feel free to join.

I think we can convert any `ElementUI` component to `Vuetify`, but I'm not sure we'll make the right decisions.

If you have design experience, we would love to have you lead the app design. (It means general design as you described, and answers questions about specific dilemmas)

You might run the app in your development environment and create a PR with design improvements.
You also can take bugs and other issues, but since you commented on this UI/UX issue, I'm talking only about UI/UX with you, that's all I know 😉

**Comment by baruchiro** [comment]
You can be the **English** code owner, too 😂
We need someone to review our English in the **app** and **documentation** (with these comments you will have to deal...)

**Comment by baruchiro** [comment]
@ChenAlon We are moving to another project, and we still need to work on the UI/UX, you are welcome to join!

See [UI Issues](https://github.com/brafdlog/budget-tracking/issues?q=is%3Aopen+is%3Aissue+label%3AUI)
